### DISCUSSION TITLE
BTT TFT35-E3 almost working with SKR 1.3

### DISCUSSION THREAD

**Discussion by systmworks** [teeminus/NoTouchScreenFirmware#50] [Q&A] [answered]
Hello,

The following config is displaying temp/fan etc information from Klipper, and clicking the encoder wheel will go into and out of the main menu.

The only thing not working is the encoder wheel left/right rotation is not moving through the submenus.
Is that movement controlled by the 2 "encoder_pins" values ?

[board_pins]
aliases:
    EXP1_1=P1.30, EXP1_3=P1.18, EXP1_5=P1.20, EXP1_7=P1.22, EXP1_9=<GND>,
    EXP1_2=P0.28, EXP1_4=P1.19, EXP1_6=P1.21, EXP1_8=P1.23, EXP1_10=<5V>,
    EXP2_1=P0.17, EXP2_3=P3.26, EXP2_5=P3.25, EXP2_7=P1.31, EXP2_9=<GND>,
    EXP2_2=P0.15, EXP2_4=P0.16, EXP2_6=P0.18, EXP2_8=<RST>, EXP2_10=<NC>

[display]
lcd_type: st7920 
cs_pin: EXP1_4
sclk_pin: EXP1_5
sid_pin: EXP1_3
encoder_pins: ^EXP2_3, ^EXP2_5
click_pin: ^!EXP1_2

**Comment by teeminus** [comment, marked as answer]
Yes, that's correct. Do you have the EXP2 cable connected?

**Reply by systmworks** [reply]
OK, I silly, did not have EXP2 cable connected (had to go looking for 2nd cable to use).
Shame to need a 2nd cable just for those 2 pins.

Had to swap the encoder pins around to get the menu moving in the correct direction, but can now navigate through the menus and select actions like Home All  etc.

I have noticed sometimes the screen will lock up and I need to power the printer/screen off and on again to get it working - have not tried printing yet as Im still getting the Klipper Z offset set correctly, as the auto mesh using the probe set the nozzle too close.

EDIT:  found the TFT35-E3 on SKR1.3 works better using the emulated display driver.    Updated config below.

[board_pins]
aliases:
    # EXP1 header
    EXP1_1=P1.30, EXP1_3=P1.18, EXP1_5=P1.20, EXP1_7=P1.22, EXP1_9=<GND>,
    EXP1_2=P0.28, EXP1_4=P1.19, EXP1_6=P1.21, EXP1_8=P1.23, EXP1_10=<5V>,
    # EXP2 header
    EXP2_1=P0.17, EXP2_3=P3.26, EXP2_5=P3.25, EXP2_7=P1.31, EXP2_9=<GND>,
    EXP2_2=P0.15, EXP2_4=P0.16, EXP2_6=P0.18, EXP2_8=<RST>, EXP2_10=<NC>
    # Pins EXP2_1, EXP2_6, EXP2_2 are also MISO, MOSI, SCK of bus "ssp0"

[display]
lcd_type: emulated_st7920
en_pin: EXP1_4
spi_software_sclk_pin: EXP1_5
spi_software_mosi_pin: EXP1_3
spi_software_miso_pin: EXP1_6
encoder_pins: ^EXP2_5, ^EXP2_3
click_pin: ^!EXP1_2
